### PR TITLE
CKAN 2.9 / Python 3

### PR DESCRIPTION
**Pull request by amercader**
Putting this out there in case someone wants to pick it up. I sadly run out of time before completion but I'd say that the py3/CKAN 2.9 support is 80% there, with still some things to iron out.

### CODE REVIEW

File: ckanext/validation/plugin/__init__.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import cgi
 import json
 
+import six
+from werkzeug.datastructures import FileStorage as FlaskFileStorage
```
Comment (ThrawnCA) [inline ckanext/validation/plugin/__init__.py:8]: Won't this be incompatible with CKAN < 2.8?

File: ckanext/validation/plugin/pylons_plugin.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    p.implements(p.IRoutes, inherit=True)
+
+    # IRoutes
+
+    def before_map(self, map_):
+
+        controller = u'ckanext.validation.controller:ValidationController'
+
+        map_.connect(
+            u'validation_read',
```
Comment (ThrawnCA) [inline ckanext/validation/plugin/pylons_plugin.py:14]: Is there a consistent named route that templates can use to work with either Flask or Pylons without alteration?

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

**Comment by ThrawnCA** [comment]
https://github.com/frictionlessdata/ckanext-validation/pull/56 fixes some of the errors but there are other problems that I'm unsure how to resolve, such as 'enqueue_job' not being called.

**Comment by ThrawnCA** [comment]
Our fork and Keitaro's (qld-gov-au and keitaroinc) now have Py3 compatibility, if you want to use them as a reference.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ jobs:
     needs: lint
     strategy:
       matrix:
-        ckan-version: [2.9, 2.9-py2, 2.8, 2.7]
```
Comment (ThrawnCA) [inline .github/workflows/test.yml]: The keitaroinc and qld-gov-au forks are still successfully running tests on 2.8.

Comment (amercader) [reply .github/workflows/test.yml]: thanks, I ported some of your changes to support the 2.8 tests

**Pull request merged**